proxy/m2_actions.c

```diff
@@ -317,18 +317,27 @@ _dump_json_prefixes (GString *gstr, GTree *tree_prefixes)
 static void
 _dump_json_properties (GString *gstr, GTree *properties)
 {
-	g_string_append_static (gstr, "\"properties\":{");
-	if (properties) {
-		gboolean first = TRUE;
-		gboolean _func (gpointer k, gpointer v, gpointer i) {
-			(void) i;
-			COMA(gstr,first);
-			oio_str_gstring_append_json_pair (gstr, (const char *)k, (const char *)v);
-			return FALSE;
+	gboolean first = TRUE;
+	gboolean _func (gpointer k, gpointer v, gpointer i) {
+		gboolean system = GPOINTER_TO_INT(i);
+		if (system ^ g_str_has_prefix((gchar*)k, "user.")) {
```

Comment:
I like this trick but it is hard to understand at a glance.
I suggest to make it a bit clearer, e.g. with a side-variable that will also make the subsequent `if (!system)` check clearer.
```
gboolean want_system = GPOINTER_TO_INT(i);
gboolean is_user = g_str_has_prefix((gchar*)k, "user.");
if (want_system ^ is_user) {
    if (is_user) /* ... */
}